defaultNatsService.go

```diff
@@ -0,0 +1,409 @@
+package natsio
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	"github.com/orchestd/dependencybundler/interfaces/configuration"
+	"github.com/orchestd/dependencybundler/interfaces/credentials"
+	"github.com/orchestd/dependencybundler/interfaces/log"
+	. "github.com/orchestd/servicereply"
+	"github.com/orchestd/servicereply/status"
+	"go.uber.org/fx"
+	"reflect"
+	"time"
+)
+
+var connect = func(natsUrl string, options ...nats.Option) (NatsConnection, error) {
+	return nats.Connect(natsUrl, options...)
+}
+
```

Comment:
pls add a comment here:
/*
This allows a 2 phase intialization:
Usage:
natsSvc := NewNatsServiceWithoutConnection(logger)
...
err:=natsSvc.Connect
*/